stdlib/public/core/EmbeddedRuntime.swift

```diff
@@ -275,3 +275,9 @@ public func swift_deletedMethodError() -> Never {
 public func swift_willThrow() throws {
 }
 
+@_extern(c, "arc4random_buf")
+public func arc4random_buf(buf: UnsafeMutableRawPointer, nbytes: Int)
+
+public func swift_stdlib_random(_ buf: UnsafeMutableRawPointer, _ nbytes: Int) {
```

Comment:
are these functions weak? e.g. if I implement swift_stdlib_random in a platform definition to a non-cryptographic-secure-pseudo-random source but want to not claim arc4random_buf is implemented would that work?

Reply:
This is a good point, and I was previously thinking that we would just have Embedded Swift have a small fixed set of well-known basically-always-present-anyway dependencies, but I do see your point that it would really be useful to have more influence and control over things like this.

I don't think Swift has a notion of weak / override-able functions... does it? Though TBH even if it did, the linker based weak function replacement is arguably quite non-obvious and "too magical". I was thinking that an explicit solution would be better, something along the lines of:

```
public enum EmbeddedConfiguration {
  // default is arc4random_buf...
  public static var randomnessSource: (@convention(c) (UnsafeMutableRawPointer, Int) -> ()) = arc4random_buf
}

public func swift_stdlib_random(_ buf: UnsafeMutableRawPointer, _ nbytes: Int) {
  EmbeddedConfiguration.randomnessSource(buf, nbytes)
}

// ...but the user can change it by simply doing this in the firmware startup code:
EmbeddedConfiguration.randomnessSource = my_other_randomness_provider
```